src/components/can_cooperation/src/commands/on_interior_vehicle_data_notification.cc

```diff
@@ -0,0 +1,120 @@
+ CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "can_cooperation/commands/on_interior_vehicle_data_notification.h"
+#include <algorithm>
+#include <vector>
+#include "json/json.h"
+#include "can_cooperation/can_module_constants.h"
+#include "can_cooperation/message_helper.h"
+#include "can_cooperation/can_module.h"
+#include "can_cooperation/can_app_extension.h"
+
+namespace can_cooperation {
+
+namespace commands {
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "OnInteriorVehicleDataNotification")
+
+OnInteriorVehicleDataNotification::OnInteriorVehicleDataNotification(
+    const application_manager::MessagePtr& message,
+    CANModuleInterface& can_module)
+    : BaseCommandNotification(message, can_module) {}
+
+OnInteriorVehicleDataNotification::~OnInteriorVehicleDataNotification() {}
+
+void OnInteriorVehicleDataNotification::Execute() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  Json::Value moduleDescription(Json::ValueType::objectValue);
+  Json::Value json;
+
+  application_manager::MessagePtr msg = message();
+  if (!service_->ValidateMessageBySchema(*msg)) {
+      LOG4CXX_ERROR(logger_, "Message validation failed");
+      return;
+  }
+
+  json = MessageHelper::StringToValue(msg->json_message());
+
+  moduleDescription[message_params::kModuleType] =
+      json[message_params::kModuleData][message_params::kModuleType];
+
+  std::vector<application_manager::ApplicationSharedPtr> applications =
+      service_->GetApplications(can_module_.GetModuleID());
+
+  for (size_t i = 0; i < applications.size(); ++i) {
+    CANAppExtensionPtr extension =
+        application_manager::AppExtensionPtr::static_pointer_cast<
+            CANAppExtension>(
+            applications[i]->QueryInterface(can_module_.GetModuleID()));
+    DCHECK(extension);
+
+    if (extension->IsSubscibedToInteriorVehicleData(moduleDescription)) {
+      msg->set_connection_key(applications[i]->app_id());
+      application_manager::MessagePtr message(
+          new application_manager::Message(*msg));
+      NotifyOneApplication(message);
+    }
+  }
+}
+
+
+std::string OnInteriorVehicleDataNotification::ModuleType(
+    const Json::Value& message) {
+  return message.get(message_params::kModuleData,
+                     Json::Value(Json::objectValue))
+      .get(message_params::kModuleType, Json::Value(""))
+      .asString();
+}
+
+
+std::vector<std::string> OnInteriorVehicleDataNotification::ControlData(
+    const Json::Value& message) {
+  Json::Value data =
+      message.get(message_params::kModuleData, Json::Value(Json::objectValue));
+  const char* name_control_data;
```

Comment:
@LuxoftAKutsan I would move it down 3 lines below. To the place before its first usage.